src/messaging.cc

```diff
@@ -283,10 +283,7 @@ bool MessageEncoder::encode_byte_array(ByteArray* object) {
     too_many_externals_ = true;
     return false;
   }
-  if (encoding_for_size()) {
-    externals_[externals_count_] = object;
-  }
-  externals_count_++;
+  externals_[externals_count_++] = object;
```

Comment:
Alternatively we could store null in there if we're encoding for size.

```suggestion
  externals_[externals_count_++] = encoding_for_size() ? null : object;
```